shim_maker.py

```diff
@@ -1,108 +1,162 @@
+        else:
+            return hasattr(member, "__module__") and member.__module__ == code.__name__
+
+    for name, member in inspect.getmembers(code, predicate=_filter_members):
+        if name.startswith("_"):
+            continue
+        if is_init:
+            imports.add(member.__name__[len(code.__name__) + 1 :])
+        else:
+            imports.add(name)
+
+    if hasattr(code, "__all__") and not is_init:
+        imports.update(code.__all__)
+
+        if len(code.__all__) > 1:
+            shim += "__all__ = (\n"
+            shim += textwrap.indent(
+                "\n".join([f'"{member}",' for member in sorted(code.__all__)]), "    "
+            )
+            shim += "\n)\n\n"
+        elif len(code.__all__) == 1:
+            shim += "__all__ = (" + f'"{code.__all__[0]}",' + ")\n\n"
+        else:
+            shim += "__all__ = ()\n\n"
+
+    if is_init:
+        shim += f"from {code.__name__} import *\n"
+        for mod in imports:
+            shim += f"from .{mod} import *\n"
+    else:
+        shim += f"from {code.__name__} import {', '.join(sorted(imports))}\n"
+
+    shim = isort.api.sort_code_string(shim, file_path=ISORT_CONFIG)
+    return shim.strip() + "\n"
+
+
+def _shim_module_type(
+    base_module: ModuleType, shim_module: str, *, original_shim: pathlib.Path = None
+):
+    """
+    Shim a module into the other location.
+    """
+    if not base_module.__file__:
+        raise RuntimeError(f"{base_module}'s __file__ attribute is not set")
+    # convert the shim_module to a path
+    shim_path = shim_module.replace(".", os.sep)
+    # handle __init__
+    if base_module.__file__.endswith("__init__.py"):
+        shim_path += os.sep + "__init__.pyi"
+        is_init = True
+    else:
+        shim_path += ".pyi"
+        is_init = False
+
+    if not os.path.exists(os.path.dirname(shim_path)):
+        os.makedirs(os.path.dirname(shim_path))
+
+    # actually shim the file
+    code = _type_shim_code(base_module, is_init=is_init)
+    with open(shim_path, "w") as f:
+        f.write(code)
+
+    if os.path.exists(py_typed := os.path.dirname(base_module.__file__) + os.sep + "py.typed"):
+        shutil.copyfile(py_typed, os.path.dirname(shim_path) + os.sep + "py.typed")
+
+
+def shim_module(base_name, shim_name, module_name, original_shim: pathlib.Path = None):
+    """Shim a module and all of its submodules."""
+    base = importlib.import_module(module_name or base_name, base_name)
+    _shim_module_type(base, shim_name + module_name, original_shim=original_shim)
+    for module in pkgutil.iter_modules(base.__path__):
+        if module.ispkg:
+            continue
+        submodule_name = module_name + "." + module.name
+        _shim_module_type(
+            importlib.import_module(base_name + submodule_name),
+            shim_name + submodule_name,
+            original_shim=original_shim and original_shim / f"{module.name}.py",
+        )
+
+
+def main(base_name: str, shim_name: str):
```

Comment:
```suggestion
def main(base_name: str, shim_name: str) -> None:
```
or alternatively `Any`

Reply:
https://github.com/DisnakeDev/discord-disnake/pull/3/commits/951fd6e4b2d574238926f123d68b9231ac8f225f